sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb

```diff
@@ -0,0 +1,239 @@
+    "\n",
+    "Script Mode supports training with ``Python`` script, ``Python`` module and shell script. In this example we will use a ``Python`` script to train the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). In addition this notebook also demonstrates how to perform real time inference with [SageMaker TensorFlow serving containers](https://github.com/aws/sagemaker-tensorflow-serving-container). TensorFlow serving container is the only supported inference method for Script Mode. For full documention on TensorFlow serving please visit [here](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/deploying_tensorflow_serving.rst).\n",
+    "\n",
+    "Let's start by setting up the environment:\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up the environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "role = 'SageMakerRole'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training Data\n",
+    "\n",
+    "The MNIST dataset has been loaded to public S3 buckets ``sagemaker-sample-data-<REGION>`` under prefix ``tensorflow/mnist``. There are four ``.npy`` file under this prefix: ``train_data.npy``, ``eval_data.npy``, ``train_labels.npy`` and ``eval_labels.npy``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_data_uri = 's3://sagemaker-sample-data-us-west-2/tensorflow/mnist'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Construct a script for distributed training \n",
+    "\n",
+    "The training script was adapted from TensorFlow's official [CNN MNIST example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py). We have modified to handle a ``model_dir`` parameter passed in by SageMaker. This is a S3 path which can be used for data sharing during distributed training and checkpoint and or model persistence. In addtion we have also added a argument parsing function to handle processing training related variables.\n",
+    "\n",
+    "At the end of the training job we also added a step to export the trained model to the path stored in the environment variable ``SM_MODEL_DIR`` which always points to ``/opt/ml/model``. This is critical because SageMaker uploads all the model artifact in this folder to S3 at end of training.\n",
+    "\n",
+    "Here is the entire script:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!cat 'mnist.py'"
```

Comment:
personally I like using `pygmentize` instead of `cat` for the syntax highlighting